src/lib/coapcore/src/ace.rs

```diff
@@ -397,3 +447,95 @@ pub fn process_acecbor_authz_info<Scope>(
 
     Ok((response, scope, derived))
 }
+
+pub fn process_edhoc_token<Scope>(
+    ead3: &[u8],
+    authorities: &impl crate::seccfg::ServerSecurityConfig<Scope = Scope>,
+) -> Result<(lakers::Credential, Scope), CredentialError> {
+    let mut buffer = heapless::Vec::<u8, MAX_SUPPORTED_ACCESSTOKEN_LEN>::new();
+
+    // Trying and falling back means that the minicbor error is not too great ("Expected tag 16"
+    // rather than "Expected tag 16 or 18"), but we don't
+    // show much of that anyway.
+    let (scope, claims) = if let Ok(encrypt0) = minicbor::decode::<EncryptedCwt>(ead3) {
+        let (headers, aad_encoded, buffer) = encrypt0.prepare_decryption(&mut buffer)?;
+
+        authorities.decrypt_symmetric_token(
+            &headers,
+            aad_encoded.as_ref(),
+            buffer,
+            crate::PrivateMethod,
+        )?
+    } else if let Ok(sign1) = minicbor::decode::<SignedCwt>(ead3) {
+        let protected: HeaderMap = minicbor::decode(sign1.protected)?;
+        trace!(
+            "Decoded protected header map {:?} inside sign1 container {:?}",
+            &protected,
+            &sign1
+        );
+        let headers = sign1.unprotected.updated_with(protected);
+
+        #[derive(minicbor::Encode)]
+        struct SigStructureForSignature1<'a> {
+            #[n(0)]
+            context: &'static str,
+            #[cbor(b(1), with = "minicbor::bytes")]
+            body_protected: &'a [u8],
+            #[cbor(b(2), with = "minicbor::bytes")]
+            external_aad: &'a [u8],
+            #[cbor(b(3), with = "minicbor::bytes")]
+            payload: &'a [u8],
+        }
+        let aad = SigStructureForSignature1 {
+            context: "Signature1",
+            body_protected: sign1.protected,
+            external_aad: &[],
+            payload: sign1.payload,
+        };
+        buffer = heapless::Vec::new();
+        minicbor::encode(&aad, minicbor_adapters::WriteToHeapless(&mut buffer))?;
+        trace!("Serialized AAD: {:#02x}", buffer);
+
+        let (scope, claims) = authorities.verify_asymmetric_token(
+            &headers,
+            &buffer,
+            sign1.signature,
+            sign1.payload,
+            crate::PrivateMethod,
+        )?;
+
+        (scope, claims)
+    } else {
+        return Err(CredentialErrorDetail::UnsupportedExtension.into());
+    };
+
+    // FIXME: Check iat/exp against raytime
```

Comment:
Addressing this is out of scope for now: We don't have trusted wall time on our system yet at all.